unity/Assets/QuestNav/Native/NTCore/StringSubscriber.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Text;
+
+namespace QuestNav.Native.NTCore
+{
+    public class StringSubscriber
+    {
+        private readonly uint handle;
+
+        internal StringSubscriber(uint handle)
+        {
+            this.handle = handle;
+        }
+
+        public unsafe string Get(string defaultValue)
+        {
+            byte[] valueUtf8 = defaultValue is not null
+                ? Encoding.UTF8.GetBytes(defaultValue)
+                : Array.Empty<byte>();
+
+            string result = null;
+            fixed (byte* ptr = valueUtf8)
+            {
+                WpiString defaultWpi = new WpiString { str = ptr, len = (UIntPtr)valueUtf8.Length };
+                WpiString outValue = new WpiString();
+                NtCoreNatives.NT_GetString(handle, &defaultWpi, &outValue);
+
+                if (outValue.str == defaultWpi.str)
```

Comment:
I haven't looked at the WPIlib code to see how it handles the default case, but if it hands the original `defaultValue` pointer back to us calling `NT_FreeRaw` would throw an exception.

If it copies our default value into a new string that will be handled below.